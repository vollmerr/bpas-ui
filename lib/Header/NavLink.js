var _templateObject = _taggedTemplateLiteral(['\n    text-align: ', ';\n    flex: 1;\n    list-style-type: none;\n\n    &:not(:last-child) {\n      border-bottom: ', ';\n    }\n'], ['\n    text-align: ', ';\n    flex: 1;\n    list-style-type: none;\n\n    &:not(:last-child) {\n      border-bottom: ', ';\n    }\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n    display: ', ';\n    vertical-align: ', ';\n    color: ', ';\n    padding-right: ', ';\n'], ['\n    display: ', ';\n    vertical-align: ', ';\n    color: ', ';\n    padding-right: ', ';\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n    display: block;\n    color: inherit;\n    padding: ', 'px ', 'px;\n\n    &:hover,\n    &:focus,\n    &.active,\n    &:hover > ', ',\n    &:focus > ', ',\n    &.active > ', ' {\n      text-decoration: none;\n      color: ', ';\n    }\n'], ['\n    display: block;\n    color: inherit;\n    padding: ', 'px ', 'px;\n\n    &:hover,\n    &:focus,\n    &.active,\n    &:hover > ', ',\n    &:focus > ', ',\n    &.active > ', ' {\n      text-decoration: none;\n      color: ', ';\n    }\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import { NavLink as Link } from 'react-router-dom';
import styled from 'styled-components';
import Icon from '../Icon';
import theme from '../util/theme';

var Wrapper = styled.li(_templateObject, function (props) {
  return props.isMobile ? 'left' : 'center';
}, function (props) {
  return props.isMobile ? '1px solid ' + theme.color.theme.tertiary : 'none';
});

var StyledIcon = styled(Icon)(_templateObject2, function (props) {
  return props.isMobile ? 'inline-block' : 'block';
}, function (props) {
  return props.isMobile ? 'middle' : 'baseline';
}, theme.color.theme.tertiary, function (props) {
  return props.isMobile ? '10px' : '0';
});

var StyledLink = styled(Link)(_templateObject3, theme.size.sm, theme.size.xl, StyledIcon, StyledIcon, StyledIcon, theme.color.theme.primary);

/** Renders a single navbar link */
function NavLink(_ref) {
  var link = _ref.link,
      isMobile = _ref.isMobile,
      onClick = _ref.onClick;

  return React.createElement(
    Wrapper,
    { isMobile: isMobile },
    React.createElement(
      StyledLink,
      {
        exact: link.href === '/',
        to: link.href,
        onClick: onClick
      },
      React.createElement(StyledIcon, {
        icon: link.icon,
        size: 32,
        isMobile: isMobile
      }),
      link.text
    )
  );
}

NavLink.PropTypes = {
  link: PropTypes.shape({
    text: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired
  }).isRequired,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func
};

export default NavLink;