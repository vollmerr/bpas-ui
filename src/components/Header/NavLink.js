import React from 'react';
import PropTypes from 'prop-types';
import { NavLink as Link } from 'react-router-dom';
import styled from 'styled-components';
import Icon from '../Icon';
import theme from '../../util/theme';

const Wrapper = styled.li`
    text-align: ${props => props.isMobile ? 'left' : 'center'};
    flex: 1;

    &:not(:last-child) {
      border-bottom: ${props => props.isMobile ? `1px solid ${theme.color.primaryFaded}` : 'none'};
    }
`;

const StyledIcon = styled(Icon)`
    display: ${props => props.isMobile ? 'inline-block' : 'block'};
    vertical-align: ${props => props.isMobile ? 'middle' : 'baseline'};
    color: ${theme.color.primaryFaded};
    padding-right: ${props => props.isMobile ? '10px' : '0'};
`;

const StyledLink = styled(Link)`
    display: block;
    color: inherit;
    padding: ${theme.size.sm}px ${theme.size.xl}px;

    &:hover,
    &:focus,
    &.active,
    &:hover > ${StyledIcon},
    &:focus > ${StyledIcon},
    &.active > ${StyledIcon} {
      text-decoration: none;
      color: ${theme.color.primary};
    }
`;

/** Renders a single navbar link */
function NavLink({
  link,
  isMobile,
  onClick,
}) {
  return (
    <Wrapper isMobile={isMobile}>
      <StyledLink
        exact={link.href === '/'}
        to={link.href}
        onClick={onClick}
      >
        <StyledIcon
          icon={link.icon}
          size={32}
          isMobile={isMobile}
        />
        {link.text}
      </StyledLink>
    </Wrapper>
  );
}

NavLink.PropTypes = {
  link: PropTypes.shape({
    text: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
  }).isRequired,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func,
};

export default NavLink;
