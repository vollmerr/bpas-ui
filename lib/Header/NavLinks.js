var _templateObject = _taggedTemplateLiteral(['\n  display: ', ';\n  justify-content: space-between;\n  list-style-type: none;\n  width: ', ';;\n  margin: 0;\n  padding: 0;\n'], ['\n  display: ', ';\n  justify-content: space-between;\n  list-style-type: none;\n  width: ', ';;\n  margin: 0;\n  padding: 0;\n']);

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import NavLink from './NavLink';

var defaultLinks = [{ text: 'Home', href: '/', icon: 'home' }, { text: 'Help', href: '/help', icon: 'question-line' }];

var Wrapper = styled.ul(_templateObject, function (props) {
  return props.isMobile ? 'block' : 'flex';
}, function (props) {
  return props.isMobile ? '100%' : '40%';
});

/** Renders the navbar links */
function NavLinks(_ref) {
  var _ref$links = _ref.links,
      links = _ref$links === undefined ? defaultLinks : _ref$links,
      _ref$isMobile = _ref.isMobile,
      isMobile = _ref$isMobile === undefined ? false : _ref$isMobile,
      props = _objectWithoutProperties(_ref, ['links', 'isMobile']);

  return React.createElement(
    Wrapper,
    { isMobile: isMobile },
    links.map(function (link) {
      return React.createElement(NavLink, Object.assign({
        link: link,
        key: link.text,
        isMobile: isMobile
      }, props));
    })
  );
}

NavLinks.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
    href: PropTypes.string,
    icon: PropTypes.string
  })),
  isMobile: PropTypes.bool
};

export default NavLinks;