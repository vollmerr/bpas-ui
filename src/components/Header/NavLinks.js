import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import NavLink from './NavLink';
import theme from '../../util/theme';

const defaultLinks = [
  { text: 'Home', href: '/', icon: 'home' },
  { text: 'Help', href: '/help', icon: 'question-line' },
];

const Wrapper = styled.ul`
  display: ${props => props.isMobile ? 'block' : 'flex'};
  flex: 1;
  justify-content: space-around;
  list-style-type: none;
  margin: 0;
  padding: 0;
  z-index: 1000;
`;

/** Renders the navbar links */
function NavLinks({
  links = defaultLinks,
  isMobile = false,
  ...props,
}) {
  return (
    <Wrapper isMobile={isMobile}>
      {links.map(link => (
        <NavLink
          link={link}
          key={link.text}
          isMobile={isMobile}
          {...props}
        />
      ))}
    </Wrapper>
  );
}

NavLinks.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      href: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
  isMobile: PropTypes.bool,
};

export default NavLinks;
