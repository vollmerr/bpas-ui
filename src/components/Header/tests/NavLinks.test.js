import React from 'react';
import { shallow } from 'enzyme';
import NavLinks from '../NavLinks';
import NavLink from '../NavLink';

const testLinks = [
    { text: 'TEST 1', href: '/sada', icon: 'phone' },
    { text: 'test 2', href: '/', icon: 'capitol' },
    { text: 'test 3', href: '/as', icon: 'menu' },
    { text: '$%^&*()', href: '/stuff-asd', icon: 'logo' },
];

describe('<NavLinks />', () => {
  let actual;
  beforeEach(() => {
    actual = shallow(
      <NavLinks />,
      { context: {history: {}} }
    );
  });

  it('renders correctly', () => {
    expect(actual).toMatchSnapshot();

    actual.setProps({links: testLinks});
    expect(actual).toMatchSnapshot();
  });

  it('maps over links passed in', () => {
    expect(actual.find(NavLink).length).toEqual(2);

    actual.setProps({links: testLinks});
    expect(actual.find(NavLink).length).toEqual(4);
  });
});
