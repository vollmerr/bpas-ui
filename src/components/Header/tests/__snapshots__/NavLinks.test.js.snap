// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> renders correctly 1`] = `
<styled.ul
  isMobile={false}
>
  <NavLink
    isMobile={false}
    link={
      Object {
        "href": "/",
        "icon": "home",
        "text": "Home",
      }
    }
  />
  <NavLink
    isMobile={false}
    link={
      Object {
        "href": "/help",
        "icon": "question-line",
        "text": "Help",
      }
    }
  />
</styled.ul>
`;

exports[`<NavLinks /> renders correctly 2`] = `
<styled.ul
  isMobile={false}
>
  <NavLink
    isMobile={false}
    link={
      Object {
        "href": "/sada",
        "icon": "phone",
        "text": "TEST 1",
      }
    }
  />
  <NavLink
    isMobile={false}
    link={
      Object {
        "href": "/",
        "icon": "capitol",
        "text": "test 2",
      }
    }
  />
  <NavLink
    isMobile={false}
    link={
      Object {
        "href": "/as",
        "icon": "menu",
        "text": "test 3",
      }
    }
  />
  <NavLink
    isMobile={false}
    link={
      Object {
        "href": "/stuff-asd",
        "icon": "logo",
        "text": "$%^&*()",
      }
    }
  />
</styled.ul>
`;
